pkg/osutil/osutil.go

```diff
@@ -157,3 +157,66 @@ func replaceTemplateVariables(fileSys fs.FS, srcPath string, draftConfig *config
 
 	return []byte(fileString), nil
 }
+
+// CopyDirWithTemplates - Handles Gotemplate processing and writing
+func CopyDirWithTemplates(
```

Comment:
Question: From what I understand, we're just replacing draftconfig with a map of varibales and values in this new func. How does that help in handling go templates? Can we not use built-in funcs from the [template](https://pkg.go.dev/text/template) package?